infra/MyStack.cs

```diff
@@ -0,0 +1,101 @@
+﻿using Pulumi;
+using Pulumi.AzureNative.Resources;
+using Pulumi.AzureNative.Web;
+using Pulumi.AzureNative.Web.Inputs;
+
+// Define global variables
+
+class MyStack : Stack
+{
+    private string CreateResourceName(params string[] args){
+        string name = "";
+        foreach (string item in args)
+        {
+            name += item + "-";
+        }
+        return name;
+    }
+
+    public MyStack()
+    {
+        // Retrieve configuration and set config variables
+        var config = new Config(); 
+        var location = config.Get("azure-native:location");
+
+        var stackName = Pulumi.Deployment.Instance.StackName;
+        var projectName = Pulumi.Deployment.Instance.ProjectName;
+
+        var appServiceSkuName = config.Require("AppServiceSkuName");
+        var staticWebAppSkuName = config.Require("StaticWebAppSkuName");
+
+        // Create an Azure Resource Group
+        var resourceGroup = new ResourceGroup(CreateResourceName("rg", projectName, stackName));
```

Comment:
just use string interpolation instead of an external method : 

```csharp
var resourceGroup = new ResourceGroup($"rg-{projectName}-{stackName}");
```